### PR TITLE
FOGL-7694 authentication and security related config items added along with group

### PR DESCRIPTION
**Pull request by ashish-jabble**
Note:
For the certificates you need to copy into FLEDGE_DATA/etc/certs location wither manually or use GUI certificate store option.
~I have made **Security** group for some config items. But the issue is we have dynamic Security category for every service and having display name "Security Configuration". So user may confuse with Security vs Security Configuration. Either GUI client should merge them internally else we need to give another name of group instead security~

### CODE REVIEW

**Review by MarkRiddoch — APPROVED** [review]

File: docs/index.rst [hunk 1]
```diff
@@ -36,20 +65,107 @@ The plugin may be used within a north *task* or *service*. Both of these are cre
...
       - **Source**: The source of the data to be sent, this may be the *readings* or *statistics* data.
 
-  - Click *Next*
+The OPC UA Security tab contains a set of configuration items that is used for setting the security between the client and the OPC UA Server.
+
++------------------------+
+| |opcuaclient_security| |
++------------------------+
+
+  - **Security Mode**: Specify the OPC/UA security mode that will be used to communicate with the OPC/UA server.
```
Comment (RVerhoeff) [inline docs/index.rst]: The terms OPC UA, OPC/UA and OPCUA all appear in the document. I suggest OPC UA everywhere.

File: docs/index.rst [hunk 2]
```diff
@@ -36,20 +65,107 @@ The plugin may be used within a north *task* or *service*. Both of these are cre
         .. code-block:: JSON
```
Comment (RVerhoeff) [inline docs/index.rst:67]: Show or explain how multiple assets and multiple datapoints in an asset appear.

File: docs/index.rst [hunk 3]
```diff
@@ -36,20 +65,107 @@ The plugin may be used within a north *task* or *service*. Both of these are cre
...
+
+    +----------------------------+
+    | |user_authentication_mode| |
+    +----------------------------+
+
+  - **Username**: Specify the username to use for authentication. This is only used if the *User Authentication Mode* is set to *Username And Password*.
+
+  - **Password**: Specify the password to use for authentication. This is only used if the *User Authentication Mode* is set to *Username And Password*.
+
+  - **Server Public Certificate**: (Optional) The name of the public certificate of the OPC/UA server specified in the *OPCUA Server URL*. This must either be a DER or PEM format certificate file.
```
Comment (RVerhoeff) [inline docs/index.rst]: For all the certificate and key files: You must include the file extension (_mycert.der_ or _mycert.pem_). Do not include the file's directory name.

File: docs/index.rst [hunk 4]
```diff
@@ -36,20 +65,107 @@ The plugin may be used within a north *task* or *service*. Both of these are cre
...
+
+- Enable your task or service
+
+- Click *Done*
+
+
+Certificate Management
+----------------------
+
+Typically the Certificate Authorities certificate is retrieved and uploaded to the Fledge Certificate Store along with the certificate from the OPC/UA server that has been signed by that Certificate Authority. A public/private key pair must also be created for the plugin and signed by the Certificate Authority. These are uploaded to the Fledge Certificate Store.
```
Comment (RVerhoeff) [inline docs/index.rst]: Explain how self-signed certificates are different. They are not signed by a Certificate Authority. The system manager must copy the server certificate to the Fledge Certificate Store, and the client certificate to the server's trusted certificate store.

File: docs/index.rst [hunk 5]
```diff
@@ -36,20 +65,107 @@ The plugin may be used within a north *task* or *service*. Both of these are cre
...
+
+|OpenSSL| may be used to generate and convert the keys and certificates required.
```
Comment (RVerhoeff) [inline docs/index.rst:177]: At the end of this paragraph, add "Copy the _ssl.conf_ file in this example to your own system when using the _openssl_ commands below to generate certificates."

File: docs/index.rst [hunk 6]
```diff
@@ -36,20 +65,107 @@ The plugin may be used within a north *task* or *service*. Both of these are cre
...
+An example to |generate_certificate| to do this is available as part of the underlying |FreeOpcUa|.
+
+.. note::
+    subjectAltName should be same as your OPCUA server Application URI.
```
Comment (RVerhoeff) [inline docs/index.rst]: Not quite correct. It should be "subjectAltName should include your OPC UA server Application URI. This URI can be created in the client certificate by setting the value of the URI token in subjectAltName in the ssl.conf file."

File: docs/index.rst [hunk 7]
```diff
@@ -36,20 +65,107 @@ The plugin may be used within a north *task* or *service*. Both of these are cre
...
+
+   openssl req -x509 -days 365 -new -out myclientcert.pem -key myclientkey.pem -config ssl.conf
+
+   openssl x509 -outform der -in myclientcert.pem -out myclientcert.der
+
+Importing the Certificate and Key Files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can either use GUI or manually to upload the certificates at right place.
+
+a) via GUI (Recommended way)
```
Comment (RVerhoeff) [inline docs/index.rst]: In this section, use "Fledge GUI" instead of just "GUI."

File: docs/index.rst [hunk 8]
```diff
@@ -36,20 +65,107 @@ The plugin may be used within a north *task* or *service*. Both of these are cre
...
+    - In the *Key* section, click *Choose File* and navigate to the location of the key file *myclientkey.pem*.
+
+    - In the *Certificate* section, click *Choose File* and navigate to the location of the certificate file *myclientcert.der*.
+
+    - Click *Import*.
+
+And for the OPC UA server certificate, enter the server certificate file name in the *Certificate* portion of the Import dialog and then click *Import*.
 
-  - Enable your task or service and click *Done*
+b) via Manually
```
Comment (RVerhoeff) [inline docs/index.rst]: No need to use "via." Say "Using the Fledge GUI" for part 1, and "Manually" for part 2.

File: docs/index.rst [hunk 9]
```diff
@@ -36,20 +65,107 @@ The plugin may be used within a north *task* or *service*. Both of these are cre
...
 
+- Copy the certificates and place in the `$FLEDGE_DATA/etc/certs/` or `$FLEDGE_ROOT/data/etc/certs/` directory. And if certificate in PEM format then place in the `$FLEDGE_DATA/etc/certs/pem/` or `$FLEDGE_ROOT/data/etc/certs/pem/` directory
```
Comment (RVerhoeff) [inline docs/index.rst]: The second sentence begins "And if certificate in PEM format then place in the…" Instead, try "If the certificate is in PEM format, place it in the…"

File: docs/index.rst [hunk 10]
```diff
@@ -23,33 +42,177 @@ The plugin may be used within a north *task* or *service*. Both of these are cre
...
+
+- Enable your task or service
+
+- Click *Done*
+
+
+Certificate Management
+----------------------
+
+Typically the Certificate Authorities certificate is retrieved and uploaded to the Fledge Certificate Store along with the certificate from the OPC UA server that has been signed by that Certificate Authority. A public/private key pair must also be created for the plugin and signed by the Certificate Authority. These are uploaded to the Fledge Certificate Store.
```
Comment (RVerhoeff) [inline docs/index.rst]: This paragraph appears to have contradictions. If I may offer an alternative: "Typically a Certificate Authority issues a client certificate and private key for the plugin which are imported into the Fledge Certificate Store along with a certificate from the OPC UA server. It is also possible to use self-signed certificates. In all cases, the system manager must copy the server certificate to the Fledge Certificate Store and the client certificate to the server's trusted certificate store. Do not copy the client private key to any other system."

**Review by RVerhoeff — CHANGES_REQUESTED** [review]
Very good. A clarification is needed for the first paragraph of Certificate Management.

**Review by RVerhoeff — APPROVED** [review]
All good!

**Pull request merged**